irods/test/query2_test.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+
+import irods.test.helpers as helpers
+
+
+class TestQuery2(unittest.TestCase):
+
+    def setUp(self):
+        self.sess = helpers.make_session()
+
+        if self.sess.server_version < (4, 3, 2):
+            self.skipTest(
+                'GenQuery2 is not available by default in iRODS before v4.3.2.')
+
+        self.coll_path_a = '/{}/home/{}/test_query2_coll_a'.format(
+            self.sess.zone, self.sess.username)
+        self.coll_path_b = '/{}/home/{}/test_query2_coll_b'.format(
+            self.sess.zone, self.sess.username)
+        self.sess.collections.create(self.coll_path_a)
+        self.sess.collections.create(self.coll_path_b)
+
+    def test_select(self):
+        query = "SELECT COLL_NAME WHERE COLL_NAME = '{}'".format(
+            self.coll_path_a)
+        q = self.sess.query2(self.sess.zone, query)
+        query_result = q.execute()
+        query_sql = q.get_sql()
+        self.assertIn([self.coll_path_a], query_result)
+        self.assertEqual(len(query_result), 1)
+        self.assertEqual(query_sql, "select distinct t0.coll_name from R_COLL_MAIN t0 inner join R_OBJT_ACCESS pcoa on t0.coll_id = pcoa.object_id inner join R_TOKN_MAIN pct on pcoa.access_type_id = pct.token_id inner join R_USER_MAIN pcu on pcoa.user_id = pcu.user_id where t0.coll_name = ? and pcoa.access_type_id >= 1000 fetch first 256 rows only")
```

Comment:
Is this Postgres? @d-w-moore may have more to say on this, but I'm wondering whether we need to make sure the database flavor on the test server is Postgres and skip as appropriate for tests like this. Is that even possible with our existing tooling?

Reply:
If postgres-specific, we could perhaps reserve a place for the test code among the motley sets of test scripts that are run under [a simplified(single-node) test harness](https://github.com/irods/python-irodsclient/pull/508), also planned for inclusion in the next release.

Reply:
I have only tested this with PostgreSQL. Testing by checking that the SQL query is exactly the same as the current result is indeed a bit fragile, I guess.

I was thinking of just checking that `get_sql` returns a string for now, or perhaps that it contains `"select"` as a sort of basic sanity check. Do you think that should be sufficient for the scope of this PR?

Reply:
I think that what you have is actually fine. I just wanted to make sure there wasn't some MySQL testing or something wherein this test may fail.

For me, at least a note saying that this is assuming Postgres would be sufficient.

More ideal would be detecting the database flavor and skipping if it's not Postgres.